Sources/DiskCache/DiskCache.swift

```diff
@@ -7,66 +7,125 @@
 
 import Foundation
 
-/// Provides interfaces for caching and retrieving data to/from disk. This implementation executes
-/// all tasks on the current queue. Consumers should manage dispatching tasks to background queues if needed
+typealias VoidCheckedContinuation = CheckedContinuation<Void, Error>
+
+/// Provides interfaces for caching and retrieving data to/from disk.
 public class DiskCache: Cache {
+    lazy var queue = DispatchQueue.global()
     let storageType: StorageType
-    let appGroupID: String?
 
-    public required init(storageType: StorageType, appGroupID: String? = nil) throws {
+    public required init(storageType: StorageType) throws {
         self.storageType = storageType
-        self.appGroupID = appGroupID
         try createDirectory(directoryURL)
     }
 
-    public func cache(_ data: Data, key: String) throws {
-        try data.write(to: fileURL(key))
+    public func cache(_ data: Data, key: String) async throws {
+        try await withCheckedThrowingContinuation { [weak self] (continuation: VoidCheckedContinuation) -> Void in
+            guard let self = self else {
+                return
```

Comment:
Would you still need to call `resume` on this code path (and all other guards within the continuation closures)?

Reply:
Long story short, I don't know...
I originally had that but removed it with the reason being I'm not sure a continuation _should_ be resumed if self nil. I don't think resuming with an error is correct in this case.

Reply:
Tried to do some basic performance tests:

```swift
// checked:
// duration: 9.661262035369873
// duration: 9.221790790557861
// duration: 9.282848119735718
//
// unsafe:
// duration: 9.469778060913086
// duration: 9.171962976455688
// duration: 9.19236707687378
func testCachePerformance() async throws {
    let cache = try DiskCache(storageType: .permanent(.custom("perm")))

    let measurementCount = 100_000
    let start = Date().timeIntervalSince1970
    for _ in 0 ..< measurementCount {
        try? await cache.cache(Self.testData, key: Self.cacheKey)
    }

    let duration = Date().timeIntervalSince1970 - start
    print("duration: \(duration)")
}
```

It would be nice if XCTest had an async `measure` method